### PR TITLE
Mon 3204 add rule for max session duration for 18 10

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

Add a rule in the administration form to require a session value lesser or equal to 120min.
Update the user's saved session value in the DB if greater to 120min

image

**Fixes** # (secu)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/Administration/parameters/general/help.php [hunk 1]
```diff
@@ -37,7 +37,13 @@
  * Sessions Properties
  */
 
-$help['tip_sessions_expiration_time'] = dgettext('help', 'Life duration of sessions.');
+$help['tip_sessions_expiration_time'] = dgettext(
+    'help',
+    'Life duration of sessions. The value in minutes cannot be greater than the session.gc_maxlifetime value set " .
```
Comment (kduret) [inline www/include/Administration/parameters/general/help.php]:
```suggestion
    'Life duration of sessions. The value in minutes cannot be greater than the session.gc_maxlifetime value set ' .
```

File: www/include/Administration/parameters/general/help.php [hunk 2]
```diff
@@ -37,7 +37,13 @@
...
+    "in your php centreon.ini file.'
```
Comment (kduret) [inline www/include/Administration/parameters/general/help.php]:
```suggestion
    'in your php centreon.ini file.'
```

File: www/include/Administration/parameters/general/help.php [hunk 3]
```diff
@@ -37,7 +37,13 @@
...
+
+
```
Comment (kduret) [inline www/include/Administration/parameters/general/help.php]: remove blank lines

File: www/include/Administration/parameters/general/form.php [hunk 4]
```diff
@@ -38,6 +38,11 @@
     exit();
 }
 
+// getting the garbage collector value set
+$lifetime = ini_get_all("session", false);
+$lifetime = ((int)$lifetime['session.gc_maxlifetime']) / 60;
+define("SESSION_DURATION_LIMIT", $lifetime);
```
Comment (ganoze) [inline www/include/Administration/parameters/general/form.php]:
```suggestion
define('SESSION_DURATION_LIMIT', (int)(ini_get('session.gc_maxlifetime') / 60));
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -14725,3 +14725,6 @@ msgstr "Filtrer par groupe de services"
 
 msgid "Chart"
 msgstr "Graphiques"
+
+msgid "This value needs to be an integer lesser than"
+msgstr "Cette valeur doit être inférieur à"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Cette valeur doit être inférieure à"
```

File: www/install/php/Update-18.10.6_to_18.10.7.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if ($sessionValue > 120) {
+        $pearDB->query(
+            'UPDATE `options` SET `value` = "120"
+            WHERE `key` = "session_expire"'
+        );
+    }
+} catch (\PDOException $e) {
+    $centreonLog->insertLog(
+        2,
+        "UPGRADE : 19.04.5 Unable to modify session expiration value"
```
Comment (kduret) [inline www/install/php/Update-18.10.6_to_18.10.7.php]:
```suggestion
        "UPGRADE : 18.10.7 Unable to modify session expiration value"
```

**Pull request merged**